var/spack/repos/builtin/packages/gromacs/package.py

```diff
@@ -128,6 +128,12 @@ def cmake_args(self):
 
         if '+mpi' in self.spec:
             options.append('-DGMX_MPI:BOOL=ON')
+            # Ensures gmxapi builds properly
+            options.extend([
+                '-DCMAKE_C_COMPILER=%s' % self.spec['mpi'].mpicc,
+                '-DCMAKE_CXX_COMPILER=%s' % self.spec['mpi'].mpicxx,
+                '-DCMAKE_Fortran_COMPILER=%s' % self.spec['mpi'].mpifc,
+            ])
```

Comment:
Why is this needed? Apparently from the build instructions `MPI` should be hooked in by just:
```python
options.append('-DGMX_MPI:BOOL=ON')
```

Reply:
Yeah, why? Seems like a bug in the gromacals build system.

@eirrgang, any idea?

Reply:
1. If GMX_MPI=ON is sufficient for the build environment, a minimal amount of compiler hinting would be preferable.
2. The recommended way to provide the MPI compiler wrappers is with hints to FindMPI.cmake rather than to the project compilers. For GROMACS 2021 and earlier, this is through MPI_C_COMPILER rather than MPI_CXX_COMPILER for historic reasons. See https://gitlab.com/gromacs/gromacs/-/issues/3671

Prior to the 2021 release, we thought that the odd bits of MPI-related CMake stuff were just weird things that could be deferred to the 2022 release, but some fixes may be necessary for the 2021 release. See also https://gitlab.com/gromacs/gromacs/-/issues/3672 https://gitlab.com/gromacs/gromacs/-/issues/3776 and https://gitlab.com/gromacs/gromacs/-/issues/3882 and please report any additional bugs.